hrs-portlets-webapp/src/main/webapp/WEB-INF/jsp/contactInfo.jsp

```diff
@@ -40,96 +40,104 @@
-                              </span>
+                          <div class='pref-name-edit'>
+                              <div class='names'>
+                                  <div class="edit-name">
+                                  <span class='label'>First</span>
+                                  <br/>
+                                  <span>
+                                      <form:input aria-label="edit first name box" path="firstName" class="uportal-input-text ${n}first-name" maxlength="30" />
+                                  </span>
+                                  </div>
+                                  <div class="edit-name">
+                                  <span class='label'>Middle</span>
+                                  <br/>
+                                  <span>
+                                      <form:input aria-label="edit middle name box" path="middleName" class="uportal-input-text ${n}middle-name" maxlength="30" />
+                                  </span>
+                                  </div>
+                                <div class="edit-name">
+                                <span class='label'>Last*</span>
+                                <br/>
+                                <span>
+                                  a  <form:input aria-label="edit last name box" path="lastName" class="uportal-input-text ${n}last-name" maxlength="30" />
+                                </span>
+                                </div>
+                              </div>
+                              <div class='info-text'>
+                                *<spring:message code='error.lastNameWeirdLogicError'/>
+                              </div>
+                              <div class="edit-buttons">
+                                  <span>
+                                      <input class="uportal-button fancy-button btn btn-primary" value="${savePreferredName}" type="submit">
+                                  </span>
+                                  <span>
+                                      <a href="#" onclick='dl_v1.displayEdit(false);' class="uportal-button fancy-cancel btn btn-default"><spring:message code="button.cancel" text="Cancel"/></a>
+                                  </span>
+                                  
+                              </div>
                               
                           </div>
-                          
                       </div>
-                  </div>
-              </spring:nestedPath>
-        </form>
+                  </spring:nestedPath>
+            </form>
+            </div>
+            <div class='edit-notice'>
+                <c:if test="${!empty prefs['notice'][0]}">
+                  <p class="prefNotice">
+                     ${prefs['notice'][0]}
+                  </p>
+                </c:if>
+            </div>
+        </c:when>
+        <c:otherwise>
```

Comment:
Thinking about how to make this code fail more gracefully, since it's been troublesome.  What if this were a  
```
<c:when test="${not empty displayName}">
```` 
or so, rather than a `<c:otherwise>`, and that way if somehow `${displayName}` was not populated for this user, would simply not show the `Primary or legal name` item.